go-runtime/ftl/reflection/type_registry.go

```diff
@@ -1,28 +1,12 @@
 package reflection
 
 import (
-	"context"
 	"reflect"
 
 	"github.com/alecthomas/types/optional"
 )
 
-type contextKeyTypeRegistry struct{}
-
-// ContextWithTypeRegistry adds a type registry to the given context.
-func ContextWithTypeRegistry(ctx context.Context, r *TypeRegistry) context.Context {
-	return context.WithValue(ctx, contextKeyTypeRegistry{}, r)
-}
-
-// TypeRegistryFromContext retrieves the [TypeRegistry] previously added to the
-// context with [ContextWithTypeRegistry].
-func TypeRegistryFromContext(ctx context.Context) optional.Option[*TypeRegistry] {
-	t, ok := ctx.Value(contextKeyTypeRegistry{}).(*TypeRegistry)
-	if ok {
-		return optional.Some(t)
-	}
-	return optional.None[*TypeRegistry]()
-}
+var singletonTypeRegistry = newTypeRegistry()
```

Comment:
Now that it is a singleton, do we need to worry more about threadsafety?
I guess all the type registration happens in `init` calls, and maybe that's all single threaded and safe?

We did something similar with ModuleContext where we now have`type Builder ModuleContext` and all the mutable funcs are only on the builder. And then when you know its no longer going to be mutated you can unwrap it back to the base type. I can imagine something similar working here if we think it's important.

Reply:
It's a great point to call out, but I think for the type registry we won't need to worry about mutability as, as you say, everything is initialised at `init()` time and after that is immutable.

Reply:
It might be worth adding a comment about the mutability guarantees though @deniseli ?

Reply:
Great points. I added the comment.